### ISSUE TITLE
regex for ignoring hidden directories doesn't work as expected

### ISSUE THREAD

**Issue by tbennett0** [PyCQA/pylint#7089]
### Bug description

In running `pylint --recursive y .` to ignore hidden directories using 2.14.4, I discovered the following pattern works:
```
[tool.pylint.MASTER]
ignore-paths = [
  "^.local/.*$"
]
```

but this one doesn't:
```
[tool.pylint.MASTER]
ignore-paths = [
  "^\\.local/.*$"
]
```

I would expect that the regex would see `.` as any character, so it matches the first, but explicitly using an escaped period that should also match doesn't cause that path to be ignored.

### Configuration

```ini
% cat .local/test.py
import blam
```

### Command used

```shell
% pylint --recursive=y .
```

### Pylint output

```shell
************* Module test
.local/test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
.local/test.py:1:0: E0401: Unable to import 'blam' (import-error)
.local/test.py:1:0: W0611: Unused import blam (unused-import)
```

### Expected behavior

No output on both examples.

### Pylint version

```shell
pylint 2.14.4
astroid 2.11.6
Python 3.9.12 (main, Jun 27 2022, 13:42:07)
[Clang 13.1.6 (clang-1316.0.21.2.5)]
```

**Comment by matusvalo** [comment]
@tbennett0 could you verify the issue with the main branch whether it fixes the issue?

**Comment by DanielNoord** [comment]
@matusvalo `2.14.4` was just released, which included your fix for #6964. So this seems to be a regression after that fix was applied.

**Comment by matusvalo** [comment]
OK I have misread the issue. So I need to fix the fix of the fix :-/. OK I am sorry for that. Doing recursive linting seems to be not so trivial as looked in the first place :-/

**Comment by DanielNoord** [comment]
Ah yeah. This is a duplicate of https://github.com/PyCQA/pylint/issues/5398.

**Comment by matusvalo** [comment]
OK so I am closing it as duplicate.
